src/backend/joanie/core/models/products.py

```diff
@@ -256,12 +203,62 @@ class Order(models.Model):
 
     class Meta:
         db_table = "joanie_order"
+        constraints = [
+            models.UniqueConstraint(
+                fields=["owner", "product"],
+                condition=~models.Q(state=enums.ORDER_STATE_CANCELED),
+                name="unique_owner_product_not_canceled",
+            )
+        ]
         verbose_name = _("Order")
         verbose_name_plural = _("Orders")
 
     def __str__(self):
         return f"Order {self.product} for user {self.owner}"
 
+    def set_enrollments(self):
+        """"Create an enrollment for each course run related to the order."""
+        enrollments = []
+        for course_run in self.course_runs.all():
+            enrollments.append(
+                Enrollment.objects.create(
+                    course_run=course_run, order=self, owner=self.owner
+                )
+            )
+        for enrollment in enrollments:
+            enrollment.set()
+
+
+class OrderCourseRelation(models.Model):
+    """
+    OrderCourseRelation model allows to define position of each courses to follow
+    for an order.
+    """
+
+    course = models.ForeignKey(
+        courses_models.Course,
+        verbose_name=_("course"),
+        related_name="order_relations",
+        on_delete=models.RESTRICT,
+    )
+    order = models.ForeignKey(
+        Order,
+        verbose_name=_("order"),
+        related_name="course_relations",
+        on_delete=models.CASCADE,
+    )
+    position = models.PositiveSmallIntegerField(_("position in order"))
+
+    class Meta:
+        db_table = "joanie_order_course_relation"
+        ordering = ("position", "course")
+        unique_together = ("order", "course")
+        verbose_name = _("Course relation to an order with a position")
+        verbose_name_plural = _("Courses relations to orders with a position")
+
+    def __str__(self):
+        return f"{self.order}: {self.position}/ {self.course}]"
+
```

Comment:
I wonder if we could not add a position attribute to the enrollment model.

Reply:
Can you elaborate what would be the purpose of this field?

Reply:
Now with the big picture, this comment is useless, I misunderstood the role of OrderCourseRelation from now.